pkg/gatewayserver/scheduling/scheduler.go

```diff
@@ -66,19 +67,32 @@ type RTTs interface {
 
 // NewScheduler instantiates a new Scheduler for the given frequency plan.
 // If no time source is specified, the system time is used.
-func NewScheduler(ctx context.Context, fp *frequencyplans.FrequencyPlan, enforceDutyCycle bool, timeSource TimeSource) (*Scheduler, error) {
+func NewScheduler(ctx context.Context, fp *frequencyplans.FrequencyPlan, enforceDutyCycle bool, scheduleAnytimeDelay *time.Duration, timeSource TimeSource) (*Scheduler, error) {
+	logger := log.FromContext(ctx)
 	if timeSource == nil {
 		timeSource = SystemTimeSource
 	}
+
+	if scheduleAnytimeDelay == nil || *scheduleAnytimeDelay == 0 {
+		scheduleAnytimeDelay = &ScheduleTimeLong
+	} else if *scheduleAnytimeDelay < ScheduleTimeShort {
+		logger.WithFields(log.Fields(
+			"schedule_anytime_delay_minimum", ScheduleTimeShort.String(),
+			"schedule_anytime_delay_requested", scheduleAnytimeDelay.String(),
+		)).Warn("schedule_anytime_delay too small, default to minimum value")
```

Comment:
- Logs are not for developers; `Requested scheduling delay is too small ...`
- Warnings are reserved for infrastructural things that operators have to look at; you can use Debug or at most Info here

Reply:
Changed to Info. I think it's important to let the operator/user that their value is not being used.